cdk/src/functions/enable-securityhub.ts

```diff
@@ -0,0 +1,546 @@
+ *  and limitations under the License.
+ */
+
+import {
+  DeregisterDelegatedAdministratorCommand,
+  EnableAWSServiceAccessCommandOutput,
+  ListAccountsCommand,
+  ListDelegatedAdministratorsCommand,
+  Organizations,
+} from '@aws-sdk/client-organizations';
+import {
+  SecurityHub,
+  CreateFindingAggregatorCommand,
+  ListFindingAggregatorsCommand,
+  UpdateFindingAggregatorCommand,
+  DeleteFindingAggregatorCommand,
+  ResourceConflictException,
+  EnableSecurityHubCommand,
+  ListOrganizationAdminAccountsCommand,
+  EnableOrganizationAdminAccountCommand,
+  DisableOrganizationAdminAccountCommand,
+  UpdateOrganizationConfigurationCommand,
+  CreateMembersCommand,
+  ListMembersCommand,
+  DisassociateMembersCommand,
+  DeleteMembersCommand,
+  DescribeStandardsControlsCommandOutput,
+  DescribeStandardsControlsCommand,
+  GetEnabledStandardsCommand,
+  StandardsStatus,
+  DescribeStandardsCommand,
+  BatchEnableStandardsCommand,
+  BatchDisableStandardsCommand,
+  UpdateStandardsControlCommand,
+} from '@aws-sdk/client-securityhub';
+import { STS } from '@aws-sdk/client-sts';
+import { getCredsFromAssumeRole } from '../utils/assume-role';
+import { delay, throttlingBackOff } from '../utils/throttle';
+
+export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent) {
+  const homeRegion = event.ResourceProperties.region;
+  const adminAccountId = event.ResourceProperties.adminAccountId;
+  const secHubCrossAccountRoleArn = event.ResourceProperties.role;
+
+  const organizationsClientManagementAccount = new Organizations({ region: 'us-east-1' });
+  const securityHubClientManagementAccount = new SecurityHub();
+
+  const stsClient = new STS();
+  const creds = await getCredsFromAssumeRole(stsClient, secHubCrossAccountRoleArn, 'EnableSecurityHub');
+  const organizationsClientAuditAccount = new Organizations({ region: 'us-east-1', credentials: creds });
+  const securityHubClientAuditAccount = new SecurityHub({ credentials: creds });
+
+  switch (event.RequestType) {
+    case 'Create':
+    case 'Update':
+      await enableOrganisationAdmin(securityHubClientManagementAccount, adminAccountId, homeRegion);
+      await createFindingAggregator(securityHubClientAuditAccount);
+      await createMembers(securityHubClientAuditAccount, organizationsClientAuditAccount);
+      await enableStandards(securityHubClientAuditAccount);
+      return { Status: 'Success', StatusCode: 200 };
+    case 'Delete':
+      await disableStandards(securityHubClientAuditAccount);
+      await deleteMembers(securityHubClientAuditAccount);
+      await deleteFindingAggregator(securityHubClientAuditAccount);
+      await disableOrganisationAdmin(securityHubClientManagementAccount, organizationsClientManagementAccount, adminAccountId, homeRegion);
+      return { Status: 'Success', StatusCode: 200 };
+  }
+}
+
+async function enableStandards(securityHubClient: SecurityHub) {
+  const foundationalSecurityBestPractices = {
```

Comment:
maybe this should be moved to the top layer or made configurable?

Reply:
I was thinking of making this configurable, so put it into a dynamic SSM parameter that is read on runtime where the end-user can enable/disable security standards and the related controls via JSON object. Im not sure if how to make it future proof tho. I would need to version the json or have some schema I can extend later on

Reply:
done